### PR TITLE
Styled components upgrade

### PR DESCRIPTION
**Pull request by kotielnikov**
### Prerequisites

Please make sure you can check the following two boxes:

- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project

### Contribution Type

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue, please reference the issue id)
- [ ] New feature (non-breaking change which adds functionality, make sure to open an associated issue first)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: docs/components/ComponentsList.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import * as React from 'react';
-import { ActionLink, Expander, styled, distance, Icon, ActionLinkProps, themed, reStyled } from '../../src';
+import { ActionLink, Expander, styled, distance, Icon, ActionLinkProps } from '../../src';
 
 // @ts-ignore
 import getUrl from 'react-styleguidist/lib/utils/getUrl';
+import { themed } from '../../src/utils/styled';
```
Comment (FlorianRappl) [inline docs/components/ComponentsList.tsx]: Sure that themed is no no exported? I mean Zeisslets will / may depend on it. Same for reStyled.

File: docs/components/StyleGuideRenderer.tsx [hunk 2]
```diff
@@ -42,33 +42,34 @@ function useTheme() {
...
-          return (
-            <Layout logo={<Logo />} ribbon={<Ribbon />} {...props}>
-              {!window.location.hash ? <HomePage /> : children}
-            </Layout>
-          );
-        }}
-      />
-    </ThemeProvider>
+    <>
+      {window.location.port === '6061' && <GlobalStyle />}
```
Comment (FlorianRappl) [inline docs/components/StyleGuideRenderer.tsx]: Woot :) Haven't seen this one. Why do we need this rule? What am I missing?

File: docs/components/Wrapper.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import { ThemeProvider } from '../../src';
 
 const ThemeWrapper: React.SFC = ({ children }) => {
   const { theme } = (window as any).context;
-  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
+  return <ThemeProvider theme={theme} children={children as any} />;
```
Comment (FlorianRappl) [inline docs/components/Wrapper.tsx]: Not sure why this needs to be changed / rewritten. Both forms are equivalent and having the children as the children is the preference (only exception would be a prop forwarding where it makes more sense than going over destructuring).

File: package.json [hunk 4]
```diff
@@ -44,7 +44,7 @@
   "peerDependencies": {
     "react": "^16.8.0",
     "react-dom": "^16.8",
-    "styled-components": "^2.x"
+    "styled-components": "^4.1.3"
```
Comment (FlorianRappl) [inline package.json]: I think `^4.x` would be sufficient for the *peer dependency*. This looks more like the *dev dependency*.

File: src/utils/index.ts [hunk 5]
```diff
@@ -13,4 +13,22 @@ export * from './displayTo';
...
+  ThemeProvider,
+  ThemeConsumer,
+  ThemeContext,
+  isStyledComponent,
+  ServerStyleSheet,
+  StyleSheetManager,
+  themed,
+  /**
+   * @deprecated
+   * use `styled` instead
```
Comment (FlorianRappl) [inline src/utils/index.ts]: Hm could also be added to Pharos UI in the compat layer and removed from Precise UI.

File: docs/components/StyleGuideRenderer.tsx (hunk 2 above)
Comment (kotielnikov) [reply docs/components/StyleGuideRenderer.tsx]: do disable animations (for visual testing)

Comment (FlorianRappl) [reply docs/components/StyleGuideRenderer.tsx]: Okay thanks good to know (alternatively could have been made work with an environment variable used when building for visual testing or a query parameter with a name like `visual_testing`)!

Comment (kotielnikov) [reply docs/components/StyleGuideRenderer.tsx]: may change with the next PR, I have some improvements for visual testing

**Pull request merged**